MapboxSearch/offline/src/main/java/com/mapbox/search/offline/OfflineSearchEngineImpl.kt

```diff
@@ -146,6 +150,30 @@ internal class OfflineSearchEngineImpl(
         }
     }
 
+    override fun searchAlongRoute(
+        query: String,
+        proximity: Point,
+        route: List<Point>,
+        executor: Executor,
+        callback: OfflineSearchCallback
+    ): AsyncOperationTask {
+        val remainingRoute = TurfMisc.lineSlice(proximity, route.last(), LineString.fromLngLats(route))
```

Comment:
I need to update my PR description. The ask was to only put this in the android search sdk for the time being and to put this in the native layer at a later date.

Reply:
So in this case, it probably doesn't make sense to introduce a separate artifact as I've suggested [here](https://github.com/mapbox/mapbox-search-android/pull/201#discussion_r1527170477). Otherwise, this will be much harder to migrate to native solution later.